src/platform/plugins/shared/dashboard/public/dashboard_api/access_control_manager.ts

```diff
@@ -15,7 +15,8 @@ import { getAccessControlClient } from '../services/access_control_service';
 
 export function initializeAccessControlManager(
   savedObjectResult?: DashboardReadResponseBody,
-  savedObjectId$?: BehaviorSubject<string | undefined>
+  savedObjectId$?: BehaviorSubject<string | undefined>,
+  clearCacheFunction?: (id: string) => void
```

Comment:
Instead of injecting `clearCacheFunction`, lets just import DashboardClient. Its a singleton so no need for prop drilling

Reply:
Yes, that makes much more sense.